ginisdk/src/doc/source/guides/common-tasks.rst

```diff
@@ -31,85 +32,81 @@ The following example shows how to create a new document from a bitmap.
+one page also should be uploaded as a partial document.
 
+.. note::
 
-To create a document from a jpeg, PDF or text these have to be uploaded as a byte array as seen in the following example.
+    PDFs and UTF-8 encoded text should also be uploaded as partial documents. Even though PDFs might
+    contain multiple pages and text is "pageless", creating partial documents for these keeps your
+    interaction with Gini consistent for all the supported document types.
 
-.. code-block:: java
-
-    import net.gini.android.Gini;
-    import net.gini.android.DocumentTaskManager;
-    import net.gini.android.models.Document;
-    
-    ...
-    
-    // Assuming that `gini` is an instance of the `Gini` facade class and `bytes` is a byte array containing a PDF or UTF-8
-    // encoded text
-
-    DocumentTaskManager documentTaskManager = gini.getDocumentTaskManager();
-    documentTaskManager.createDocument(bytes, "myFirstDocument.pdf", DocumentType.INVOICE).onSuccess(new Continuation<Document, Void>() {
-                @Override
-                public Void then(Task<Document> task) throws Exception {
-                    Document document = task.getResult();
-                    Log.d("gini", "Created document with ID " + document.getId());
-                    return null;
-                }
-    });
+Extractions are not available for partial documents. Creating a partial document is analogous to an
+upload. For retrieving extractions see :ref:`getting-extractions`.
 
 .. note::
     
-    The filename (``myFirstDocument.pdf`` in the example) is not required, it could be ``null``, but setting a filename is a good praxis
-    for human readable document identification.
-
+    The filename (``myFirstDocument.jpg`` in the example) is not required, it could be ``null``, but
+    setting a filename is a good praxis for human readable document identification.
 
 Working with optional arguments
 -------------------------------
 
-You may have noticed that we used ``null`` as the argument for the document's doctype in the first example. 
-This is a completely valid thing to do since the argument is annoted with the ``Nullable``
-annotation (``@org.jetbrains.annotations.Nullable``). All methods that accept null arguments use the
-``@Nullable`` annotation for those arguments. Consider all arguments which do not have the ``@Nullable``
-annotation as mandatory. The method will raise a ``NullPointerException`` if you pass null to such
-arguments.
+You may have noticed that we used ``null`` as the argument for the document's doctype in the
+example. This is completely valid since the argument is annoted with the ``Nullable`` annotation
+(``@org.jetbrains.annotations.Nullable``). All methods that accept null arguments use the
+``@Nullable`` annotation for those arguments. Consider all arguments which do not have the
+``@Nullable`` annotation as mandatory. The method will raise a ``NullPointerException`` if you pass
```

Comment:
I expect them to have a @NonNull annotation though

Reply:
You are right. This is legacy stuff though and is shown updated due to me tidying up the formatting (line length).